### ISSUE TITLE
CCI warns about 'extra fields not permitted' even if the yaml configuration is correct  

### ISSUE THREAD

**Issue by LudoMeurillon** [SFDO-Tooling/CumulusCI#2723]
**Describe the bug**
Hi everyone, since a few cumulusci releases we are seeing verbose warnings with a 'extra fields not permitted' message.
The issue is that is seems that our cumulusci.yml is correct.

We use a project config with custom properties like this:
```yaml
project:
    name: projectName
    package:
        name: PackageName
        namespace: namespace
        api_version: 52.0
    source_format: sfdx
    git:
        default_branch: main
        repo_url: https://github.com/...
    community:
        name: MyCommunity
        prefix: sample
    default_admin_alias: UUser
```
When we use `cci` it reports warnings about 'extra fields not permitted'
```
  '/.../cumulusci.yml' -> 'project' -> 'community'
    extra fields not permitted
  '/.../cumulusci.yml' -> 'project' -> 'default_admin_alias'
    extra fields not permitted
```
We also have the same warning on fields that are not custom and that should be fully supported
```yaml
orgs:
    scratch:
        dev:
            config_file: cumulusCI/orgs/dev.json
            days: 7
            namespaced: True
            noancestors: True
        qa:
            config_file: cumulusCI/orgs/qa.json
            days: 7
            namespaced: True
            noancestors: True
```
then `cci` reports `noancestors` as an extra field also.
```
'/.../cumulusci.yml' -> 'orgs' -> 'scratch' -> 'dev' -> 'noancestors'
    extra fields not permitted
'/.../cumulusci.yml' -> 'orgs' -> 'scratch' -> 'qa' -> 'noancestors'
    extra fields not permitted
```

**Reproduction steps**
1. run `cci project info` with a `cumulusci.yml` file containing mentioned snippets

**Context**
Please include the following information at minimum.

- Your CumulusCI version (`cci version`).
CumulusCI version: 3.38.0 (/usr/local/Cellar/cumulusci/3.38.0/libexec/bin/cci)
- Your Python version (`python --version`)
Python version: 3.9.6 (/usr/local/opt/python@3.9/bin/python3.9)
- Your operating system and version.
MacOS Big Sur
- If using Windows, whether you are using Command Prompt, PowerShell, Bash on Windows, or WSL.
- If possible, whether you installed CumulusCI using `brew`, `pip`, or `pipx`.
Installed with `brew`

**Gist**
If applicable, please use the `cci error gist` command to create a Gist from your error log and include the link here. See the [documentation](https://cumulusci.readthedocs.io/en/latest/features.html#working-with-errors) to learn how to create a Gist.

**Additional information**

**Comment by davidmreed** [comment]
The issue with `noancestors` was fixed in [CumulusCI 3.39.0](https://github.com/SFDO-Tooling/CumulusCI/releases/tag/v3.39.0) last week.

More broadly, we need to define a location within the `project` hierarchy for project-specific configuration variables, which would then not be validated. However, it's important to know that we have no plans to make these warnings into errors at this time: they are completely harmless.

**Comment by davidmreed** [comment]
We are shipping the YAML location `project__custom` to store project-specific custom markup in the next release of CumulusCI. Thanks!
